src/stream/src/common/table/state_table.rs

```diff
@@ -400,7 +400,7 @@ where
         order_types: Vec<OrderType>,
         pk_indices: Vec<usize>,
         Distribution {
-            dist_key_indices,
+            dist_key_in_pk_indices,
```

Comment:
because we change the meaning here, we should change all the `dist_key_in_pk_indices` in the optimizer's `TableCatalogBuilder` or stream executors' `from_proto`